### PR TITLE
Delete: exit with non zero if one of the containers encountered an error

### PR DESCRIPTION
**Pull request by datawolf**
Signed-off-by: Wang Long <long.wanglong@huawei.com>

### CODE REVIEW

File: delete.go [hunk 1]
```diff
@@ -79,18 +82,25 @@ status of "ubuntu01" as "stopped" the following will delete resources held for
...
 					}
 				} else {
 					fmt.Fprintf(os.Stderr, "cannot delete container %s that is not stopped: %s\n", id, s)
+					hasError = true
 				}
 			}
 		}
+
+		if hasError {
+			return fmt.Errorf("all or part of the containers delete failed")
```
Comment (cyphar) [inline delete.go]: I'd prefer to write it as `"one or more of the container deletions failed"`.

**Comment by datawolf** [comment]
@cyphar update and rebase done.

**Comment by cyphar** [comment]
LGTM

**Comment by hqhq** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/opencontainers/runc/pull-request/1078/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=opencontainers/runc)

**Pull request merged**